### PR TITLE
Fix renderer issues and DI issues with dialog-controller

### PR DESCRIPTION
**Pull request by jedd-ahyoung**

### CODE REVIEW

**Comment by jedd-ahyoung** [comment]
@EisenbergEffect Do you have time to review this?

**Comment by EisenbergEffect** [comment]
I can review shortly. I'm about to work on the dialog to update it to shadow dom v1 and fix some other issues.

File: src/dialog-controller.js [hunk 1]
```diff
@@ -8,8 +8,10 @@ import {invokeLifecycle} from './lifecycle';
 export class DialogController {
   settings: any;
   constructor(renderer: DialogRenderer, settings: any, resolve: Function, reject: Function) {
+    let defaultSettings = renderer ? renderer.defaultSettings || {} : {};
```
Comment (EisenbergEffect) [inline src/dialog-controller.js]: I don't understand this. Why/How could the DialogRenderer be null/undefined?

**Comment by EisenbergEffect** [comment]
I have reviewed. I only had one question. @jedd-ahyoung Please respond asap so I can merge. I've got a lot of other work to do on this library and need to get it done asap...as in now because this library is completely broken with the latest releases due to the shadow dom update.

File: src/dialog-controller.js (hunk 1 above)
Comment (jedd-ahyoung) [inline src/dialog-controller.js]: This was related to https://github.com/aurelia/dialog/issues/157. I wasn't sure but it looked as though people were trying to manually inject the DialogController using a container that didn't have a DialogRenderer attached to it. I wasn't entirely sure about the use case, but this apparently worked before the renderer refactor (if I remember correctly, the controller didn't need a reference to the renderer, so the renderer wasn't a constructor argument). If this is a questionable change, I'm okay with removing it.

Comment (EisenbergEffect) [inline src/dialog-controller.js]: I'm going to merge and then take a look at the whole codebase and see if there's something that can be improved here. Thanks!

**Pull request merged**